### ISSUE TITLE
Migration to the Material 3 style

### ISSUE THREAD

**Issue by meenbeese** [guardianproject/orbot#984]
Upgrading our app to Material 3 is essential for a modern aesthetic, aligning with design trends. This transition enables seamless adaptation to user wallpaper colors and introduces a much-needed light mode. Addressing issues #736 and #861 is critical, as they are key steps in the comprehensive overhaul of components and libraries. Systematically reworking these elements ensures a smooth transition, creating a visually appealing and user-friendly app in line with contemporary expectations.

**Comment by meenbeese** [comment]
If I wasn't clear enough before, this should be implemented as an option that can be toggled, as most apps do. This way, users can choose to use the default colors of Orbot and not be bound to their wallpaper colors. I assume that's what your concern is?

**Comment by RokeJulianLockhart** [comment]
> https://github.com/guardianproject/orbot/issues/984#issue-1916273743

@meenbeese, https://gitlab.com/guardianproject/torservices/-/issues/6 relates.

**Comment by meenbeese** [comment]
I am gearing to do the migration like this: `AppCompat` -> `Material Components` -> `Material3`,  where we currently are at `AppCompat` and will use the next step as an intermediate one towards M3. I just need to get in a couple of fixes before I can share a beta here to get some feedback,  if anyone is up for it :)

**Comment by meenbeese** [comment]
Sharing some screenshots here to get initial feedback. Also, these are M3 screenshots since I might just skip to M3, given that implementing M3 after Material Components is only trivial work.
<div>
  <img src="https://github.com/user-attachments/assets/ea8c18d4-8f87-4255-9595-fdbfd4b08fc5" width="24%" alt="Image 1">
  <img src="https://github.com/user-attachments/assets/3cfa8163-2467-4928-8130-128d1b3b03f4" width="24%" alt="Image 2">
  <img src="https://github.com/user-attachments/assets/e40ed89d-eff6-4931-a515-73705a235a21" width="24%" alt="Image 3">
  <img src="https://github.com/user-attachments/assets/7b8d5b00-1aa1-4105-b856-eebaa4ecbb1a" width="24%" alt="Image 4">
</div>

**Comment by ch2a** [comment]
> Sharing some screenshots here to get initial feedback. Also, these are M3 screenshots since I might just skip to M3, given that implementing M3 after Material Components is only trivial work.
> <div>
>   <img src="https://github.com/user-attachments/assets/ea8c18d4-8f87-4255-9595-fdbfd4b08fc5" width="24%" alt="Image 1">
>   <img src="https://github.com/user-attachments/assets/3cfa8163-2467-4928-8130-128d1b3b03f4" width="24%" alt="Image 2">
>   <img src="https://github.com/user-attachments/assets/e40ed89d-eff6-4931-a515-73705a235a21" width="24%" alt="Image 3">
>   <img src="https://github.com/user-attachments/assets/7b8d5b00-1aa1-4105-b856-eebaa4ecbb1a" width="24%" alt="Image 4">
> </div>

I know this is just the initial design, but the colour of the switch seems off to me.

**Comment by beedell-roke** [comment]
> https://github.com/guardianproject/orbot-android/issues/984#issuecomment-2985621229

@nvcg, you're correct: it doesn't adhere to WCAG contrast guidelines, not does it utilise the correct Material You palette colour.

**Comment by bitmold** [comment]
I do like some of this design. I'm clearing out the backlog of old tickets right now so closing, but you're always welcome to submit any new UI ideas you have
